index.js

```diff
@@ -1,39 +1,342 @@
-const readlineInterface = readline.createInterface(process.stdin, process.stdout);
+/**
+ *  Week 2 BCA Project:
+ *    Zorkington - a text based adventure game played
+ *    using the console modeled after the game 'Zork'
+ *    Author: Nick Castle, Connor Greenbaum
+ *    Date: 06/13/19
+ */
+
+// readline boilerplate code
+const readline = require("readline");
+const readlineInterface = readline.createInterface(
+  process.stdin,
+  process.stdout
+);
 
 function ask(questionText) {
   return new Promise((resolve, reject) => {
     readlineInterface.question(questionText, resolve);
   });
 }
+// end readline
 
-// remember the StateMachine lecture
-// https://bootcamp.burlingtoncodeacademy.com/lessons/cs/state-machines
-let states = {
-  'roomOne': { canChangeTo: [ 'roomTwo' ] },
-  'roomTwo': { canChangeTo: [ 'roomThree' ] },
-  'roomThree': { canChangeTo: [ 'roomOne' ] }
-};
+/**** Classes  ****/
 
-let currentState = "green";
+// class to hold game state data
+class GameState {
+  constructor(userName) {
+    this.currentRoom = rooms.Main;
+    // current player
+    this.user = new Player(userName);
+  }
+
+  moveToRoom(nextRoom) {
+    this.currentRoom = rooms[nextRoom];
+  }
 
-function enterState(newState) {
-  let validTransitions = states[currentState].canChangeTo;
-  if (validTransitions.includes(newState)) {
-    currentState = newState;
+  printState() {
+    console.log(this);
+  }
+}
+
+// Room class
+class Room {
+  constructor(roomName, inventory) {
+    this.roomName = roomName;
+    this.inventory = inventory || [];
+    this.message = roomMessages[this.roomName];
+  }
+  addToInventory(item) {
+    this.inventory.push(item);
+  }
+  removeFromInventory(item) {
+    this.inventory.splice(this.inventory.indexOf(item), 1);
+  }
+  showInventory() {
+    console.log("Room Inv: " + this.inventory);
+  }
+  // if this returns true un-lock door
+  unlockDoor(input) {
+    let password = "12345";
+    if (input.includes(password)) {
+      return true;
+    }
+  }
+}
+
+// Player class
+class Player {
+  constructor(name, status, inventory) {
+    this.name = name;
+    this.inventory = inventory || [];
+    this.status = status || "Healthy";
+  }
+  checkName() {
+    return this.name;
+  }
+  checkStatus() {
+    return this.status;
+  }
+  changeStatus(status) {
+    this.status = status;
+  }
+  addToInventory(item) {
+    global.game.currentRoom.removeFromInventory(item);
+    this.inventory.push(item);
+  }
+  dropItem(item) {
+    this.inventory.splice(this.inventory.indexOf(item), 1);
+    global.game.currentRoom.addToInventory(item);
+  }
+  showInventory() { 
+    console.log(`Player Inv: ${this.inventory}`);
+  }
+}
+// end classes
+
+/**** Game Logic ****/
+
+// creates a global GameState object and starts
+// the code for the first 'room'
+async function initGame() {
+  let userName = await ask("Hello, what is your name? \n>");
+  global.game = new GameState(userName);
+  play(global.game.currentRoom);
```

Comment:
It would improve readability if you just passed rooms.Main, because your initial room never changes when you first start up the game so there's no reason to query the game object for a static value.